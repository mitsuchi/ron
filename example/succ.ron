op 50 : _ + _ = _
op 50 : succ _

0 + n = n
succ m + n = succ p {
    m + n = p
}

main {
    succ 0 + succ 0 = x
}