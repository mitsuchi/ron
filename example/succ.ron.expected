x = succ succ 0
